File: ontrack-git/src/main/java/net/nemerosa/ontrack/git/model/plot/GPlot.java
package net.nemerosa.ontrack.git.model.plot;

import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class GPlot {

    private final List<GItem> items = new ArrayList<>();

    public GPlot add(GItem item) {
        items.add(item);
        return this;
    }

    public int getWidth() {
        int width = 0;
        for (GItem item : items) {
            width = Math.max(width, item.getMaxX());
        }
        return width;
    }

    public int getHeight() {
        int height = 0;
        for (GItem item : items) {
            height = Math.max(height, item.getMaxY());
        }
        return height;
    }
}

File: ontrack-git/src/main/java/net/nemerosa/ontrack/git/model/plot/GPlot.kt
package net.nemerosa.ontrack.git.model.plot

import kotlin.math.max

class GPlot {

    val items = mutableListOf<GItem>()

    val width: Int
        get() {
            var width = 0
            for (item in items) {
                width = max(width, item.maxX)
            }
            return width
        }

    val height: Int
        get() {
            var height = 0
            for (item in items) {
                height = max(height, item.maxY)
            }
            return height
        }

    fun add(item: GItem): GPlot {
        items.add(item)
        return this
    }

}
